### PR TITLE
Switch to new docker client

### PR DESCRIPTION
**Pull request by Random-Liu**
I am working on switching Kubernetes to [new docker engine-api](https://github.com/docker/engine-api) 
See https://github.com/kubernetes/kubernetes/issues/23563.

In this PR, I switched the cadvisor to using the official engine-api as well. The engine-api only supports docker apiversion>1.2. But cadvisor only uses limited functions, all these functions are backward compatible.

@vishh @timstclair

### CODE REVIEW

**Comment by timstclair** [comment]
ok to test

**Comment by Random-Liu** [comment]
@timstclair Rebased. :)

File: container/docker/handler.go [hunk 1]
```diff
@@ -172,18 +174,23 @@ func newDockerContainerHandler(
...
+		return nil, fmt.Errorf("failed to parse the create timestamp %q for container %q: %v", ctnr.Created, id, err)
+	}
 	handler.pid = ctnr.State.Pid
 
 	// Add the name and bare ID as aliases of the container.
 	handler.aliases = append(handler.aliases, strings.TrimPrefix(ctnr.Name, "/"), id)
 	handler.labels = ctnr.Config.Labels
 	handler.image = ctnr.Config.Image
-	handler.networkMode = ctnr.HostConfig.NetworkMode
+	handler.networkMode = string(ctnr.HostConfig.NetworkMode)
```
Comment (timstclair) [inline container/docker/handler.go]: Leave this the docker NetworkMode type, then you can use `networkMode.IsContainer()` in `needNetwork`

**Comment by timstclair** [comment]
LGTM, just one small comment.

File: container/docker/handler.go (hunk 1 above)
Comment (Random-Liu) [inline container/docker/handler.go]: ACK, :0

**Comment by Random-Liu** [comment]
@timstclair Addressed comments.

**Comment by timstclair** [comment]
LGTM

**Pull request merged**